docs/reference/plugins/developing_custom_plugins.md

```diff
@@ -0,0 +1,54 @@
+# Developing Custom Plugins
+
+Shift Plugins are developed as custom *PySide2* classes inheriting from `QtWidgets.QWidget`.
+
+## Writing a Plugin Class
```

Comment:
I feel there's a missing paragraph about what is the general structure of a Plugin? Even if it is just saying that, as a general rule, it should consist on the logic to build the widget UI and then connect the respective signals. For instance, I would talk about the connectBoardSignals and its intended use at the beginning of the documentation for writing the plugin.

Then, I would indicate to the user that there's examples in the shift folder they can consult for better understanding.

Reply:
Unfortunately, there is no standard way to do that Julia. Other plugins from ftrack for example have a different method structure. As we do not have an API for plugins, I think it would be meaningless to talk to the user referring to specifically named methods. As there is not restriction on those....
I put a paragraph to explain the logic behind the suggestion of structuring things like that a little later.

Reply:
Yes, I read the paragraph. What I'm suggesting is to start with the suggested structure at the beginning of the Plug class explanation. It can be clarified that it's not a closed structure but that, generally, you would have the widget UI definition in the constructor and then (as a recommendation) a dedicated method for signals.

Reply:
done

Reply:
I think now looks great! I like how you added bullet points and a code example for each step. Thanks!